src/agent/agent_manager.py

```diff
@@ -0,0 +1,279 @@
+#from src.telemetry_handler import update_telemetry
+
+from src.agent.agent_prompts import AgentPrompts
+from src.agent.agent_runner import AgentRunner
+
+
+@singleton
+class AgentManager:
+    def __init__(self, telemetry_handler):
+        debug_log("Initializing AgentManager")
+        self.telemetry_handler = telemetry_handler
+        self.agent_runner = AgentRunner(telemetry_handler)
+
+    def _build(self, remediation_id: str, build_command: str, repo_root: str) -> Tuple[bool, str]:
+        log(f"\n--- Running Build Command: {build_command} ---")
+        try:
+            # Use shell=True if the command might contain shell operators like &&, ||, > etc.
+            # Be cautious with shell=True if the command comes from untrusted input.
+            # Here, it's from an environment variable, assumed to be controlled.
+            result = subprocess.run(
+                build_command,
+                cwd=repo_root,
+                shell=True,
+                check=False, # Don't raise exception on non-zero exit
+                capture_output=True,
+                text=True,
+                encoding='utf-8', # Explicitly set encoding
+                errors='replace' # Handle potential encoding errors in output
+            )
+            self.telemetry_handler.update_telemetry("configInfo.buildCommandRunTestsIncluded", True);
+            output = result.stdout + result.stderr
+            if result.returncode == 0:
+                log("Build command succeeded.")
+                return True, output
+            else:
+                debug_log(f"Build command failed with exit code {result.returncode}.")
+                return False, output
+        except FileNotFoundError:
+            log(f"Error: Build command '{build_command}' not found. Is it installed and in PATH?", is_error=True)
+            error_exit(remediation_id)
+        except Exception as e:
+            log(f"An unexpected error occurred while running the build command: {e}", is_error=True)
+            error_exit(remediation_id)
+
+    def _format(self, remediation_id: str, formatting_command: str, repo_root: Path) -> List[str]:
+        changed_files = []
+        if not formatting_command:
+            return changed_files
+            
+        log(f"\n--- Running Formatting Command: {formatting_command} ---")
+        # Modified to match run_command signature which returns only stdout
+        current_dir = os.getcwd()
+        try:
+            os.chdir(str(repo_root))  # Change to repo root directory
+            try:
+                format_output = run_command(
+                    formatting_command.split(),  # Split string into list for Popen
+                    check=False  # Don't exit on failure, we'll check status
+                )
+                format_success = True  # If no exception was raised, consider it successful
+            except Exception as e:
+                format_success = False
```

Comment:
If you swapped the chdir inside the other block, you could put your whole failure block in here:

```
log(f"::error::Error executing formatting command: {formatting_command}")
log(f"::error::Error details: {format_output}", is_error=True)
error_exit(remediation_id)
```